server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -401,14 +403,12 @@ public String executor() {
                 new InitializeExtensionRequest(transportService.getLocalNode(), extension),
                 initializeExtensionResponseHandler
             );
-            // TODO: make asynchronous
-            inProgressFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
-        } catch (Exception e) {
-            try {
-                throw e;
-            } catch (Exception e1) {
-                logger.error(e.toString());
+            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info("No response from extension to request.");
             }
+            throw RuntimeException.class.cast(e.getCause());
```

Comment:
We cannot cast like that, we need to check first:
```suggestion
if (e.getCause() instanceof RuntimeException) {
     throw (RuntimeException) e.getCause();
} else if (e.getCause() instanceof Error) {
     throw (Error) e.getCause();
} else {
     throw new RuntimeException(e.getCause());
}
```